### PR TITLE
[JENKINS-38680] More descriptive error messages

### PR DESCRIPTION
**Pull request by pjanouse**

### CODE REVIEW

File: src/main/java/com/redhat/foreman/ForemanSharedNodeCloud.java [hunk 1]
```diff
@@ -472,7 +472,7 @@ public FormValidation doTestConnection(@QueryParameter("url") String url,
                     return FormValidation.error(Messages.Error() + ": " + e);
                 }
             }
-            return FormValidation.error(Messages.InvalidURI());
+            return FormValidation.error("Invalid Foreman API Url");
```
Comment (scoheb) [inline src/main/java/com/redhat/foreman/ForemanSharedNodeCloud.java]: Why the move from Messages?

Comment (pjanouse) [inline src/main/java/com/redhat/foreman/ForemanSharedNodeCloud.java]: My bad of course, I thought the content of message comes from the Jenkins core, will fix it.

Anyway we should move to the catalog all messages uses for WebGUI feedback.

**Comment by pjanouse** [comment]
I'm not so happy with the key `CheckCompatibleHostsOK`, is there a better way how to handle such a case?

**Comment by scoheb** [comment]
What are you not happy with? The actual key *CheckCompatibleHostsOK*? The value in the properties file? Or the actual action of determining what is compatible?

**Comment by pjanouse** [comment]
With the value in the property file.

**Comment by scoheb** [comment]
Well...any suggestions?

We can change it to be "The following hosts are available to be used:"

Essentially, I want to let users see that the hosts that they have setup on the Foreman side are usable by the plugin.

**Comment by pjanouse** [comment]
I'm definitely for the background idea. 

Isn't there a better way how to handle HTML format and dynamically created part of the content? 
Probably some other plugin does the same where we can inspire...

**Comment by scoheb** [comment]
Maybe...at this point, I do not want the HTML format to derail or block us since it is cosmetic.

Why not file an issue and we can address it later?

**Pull request merged**